backend/chalice/api_server/app.py

```diff
@@ -49,6 +53,17 @@ def get_chalice_app(flask_app):
     app.debug = flask_app.debug
     app.log.setLevel(logging.DEBUG)
 
+    # set the flask secret key, needed for session cookies
+    flask_secret_key = "OpenSesame"
+    deployment = os.environ["DEPLOYMENT_STAGE"]
+    if deployment != "test":
+        secret_name = f"corpora/backend/{os.environ['DEPLOYMENT_STAGE']}/auth0-secret"
+        auth_secret = json.loads(AwsSecret(secret_name).value)
```

Comment:
We need a strategy to avoid calling Aws Secrets for every api request. It add an additional remote call to every authenticated endpoint. A timed cache sounds like the appropriate solution. Maybe grab the secret every 5 minutes. This would be a change made to AwsSecret.

Reply:
This doesn't need to be done in this PR.

Reply:
Agreed.   I'll do that in the next pass.

Reply:
@Bento007 was there a reason you thought it should be a timed cache, rather than just a cache.   I'm asking because a regular cache I can implement with 1 line on the "get_oauth_config" function.   I guess a reason for it to be timed is because we could then change it on the fly without redeploying - but that would lead to other issues for the up to 5 minutes when it would be out of sync.   

Also, I see there is a base class called SecretConfig which is used for CorporaConfig and CorporaDbConfig.  Those also cache permanently.  I didn't see that when I first implemented this.  It looks like I could follow along with that pattern.  Any thoughts on that?

Reply:
> but that would lead to other issues for the up to 5 minutes when it would be out of sync.

A time cache was the first solution to come to mind but we could also, redeploy the lambdas every time the secret rotates. For now we can wait and see if it become a problem.

>  I didn't see that when I first implemented this. It looks like I could follow along with that pattern. Any thoughts on that?

:+1: